staging/src/k8s.io/apimachinery/pkg/labels/selector.go

```diff
@@ -793,7 +793,7 @@ func validateLabelValue(v string) error {
 // SelectorFromSet returns a Selector which will match exactly the given Set. A
 // nil and empty Sets are considered equivalent to Everything().
 func SelectorFromSet(ls Set) Selector {
-	if ls == nil {
+	if ls == nil || len(ls) == 0 {
```

Comment:
Just FYI, it's safe to call len on nil maps so you could have just replaced the == nil with the clause you added.